src/test/java/org/broadinstitute/barclay/help/DefaultDocWorkUnitHandlerUnitTest.java

```diff
@@ -0,0 +1,89 @@
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class DefaultDocWorkUnitHandlerUnitTest {
+
+    @DataProvider
+    public static Object[][] workUnitDescriptions() {
+        return new Object[][] {
+                {TestArgumentContainer.class, "", Collections.EMPTY_MAP, ""},
+                {TestExtraDocs.class, "Javadoc description", Collections.EMPTY_MAP, ""},
+                {TestExtraDocs.class, "Javadoc description \nin two lines", Collections.singletonMap("", "Inline tag"), "Inline tag"}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] workUnitSummary() {
+        return new Object[][] {
+                {TestArgumentContainer.class, "", Collections.EMPTY_MAP, TestArgumentContainer.ONE_LINE_SUMMARY},
+                {TestExtraDocs.class, "Javadoc description", Collections.EMPTY_MAP, "Javadoc description"},
+                {TestExtraDocs.class, "Javadoc description \nin two lines", Collections.singletonMap("", "Inline tag"), "Javadoc description in two lines"}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] workUnitGroupName() {
+        return new Object[][] {
+                {TestArgumentContainer.class, TestArgumentContainer.GROUP_NAME},
+                {TestExtraDocs.class, TestExtraDocs.GROUP_NAME}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] workUnitGroupSummary() {
+        return new Object[][] {
+                {TestArgumentContainer.class, TestProgramGroup.DESCRIPTION},
+                {TestExtraDocs.class, Collections.EMPTY_MAP, ""},
+        };
+    }
+
+    @Test(dataProvider = "workUnitDescriptions")
+    public void testGetDescription(final Class<?> docWorkUnitClazz, final String javadocText, final Map<String, String> inlineTags,
+            final String expectedDescription) {
+        final DocWorkUnit mockWorkUnit = createMockWorkUnit(docWorkUnitClazz, javadocText, inlineTags);
+        Assert.assertEquals(getDefaultWorkUnitHandlerInstance().getDescription(mockWorkUnit), expectedDescription);
+    }
+
+    @Test(dataProvider = "workUnitSummary")
+    public void testGetSummaryForWorkUnit(final Class<?> docWorkUnitClazz, final String javadocText, final Map<String, String> inlineTags,
+            final String expectedSummary) {
+        final DocWorkUnit mockWorkUnit = createMockWorkUnit(docWorkUnitClazz, javadocText, inlineTags);
+        Assert.assertEquals(getDefaultWorkUnitHandlerInstance().getSummaryForWorkUnit(mockWorkUnit), expectedSummary);
+    }
+
+    @Test(dataProvider = "workUnitGroupName")
+    public void testGetGroupNameForWorkUnit(final Class<?> docWorkUnitClazz, final String expectedGroupName) {
+        final DocWorkUnit mockWorkUnit = DocGenMocks.createDocWorkUnit(
+                getDefaultWorkUnitHandlerInstance(), docWorkUnitClazz);
+        Assert.assertEquals(getDefaultWorkUnitHandlerInstance().getGroupNameForWorkUnit(mockWorkUnit), expectedGroupName);
```

Comment:
There are several of these where there are multiple calls to getDefaultWorkUnitHandlerInstance in the same test. It think there should be one DefaultWorkUnitHandler created here, which is used to both mock the workUnit and to retrieve the groupName.

The fact that this test works as is indicates that getGroupNameForWorkUnit (and getGroupSummaryForWorkUnit ) should really be methods on DocWorkUnit, not on the handler, since it appears to not matter what handler instance is used. (I think this can up in a previous PR too). But thats for another day. For now we should fix the tests to use the same one.

Reply:
I think that it shouldn't be re-used, because we should test how a state-less WorkUnitHandler behaves.

Regarding your side note, the default handler is independent of the instance used, but the user can override the behaviour. An usage case is to document a toolkit which includes another one (e.g., Picard inside GATK). In that case, a custom handler may want to group together all the Picard programs to have them in a different page or pointing that they do not belong to the custom toolkit but to a third-party. I think that this methods belongs to the handle even if the default behaviour is instance-independent...

Reply:
No action was taken here.

Reply:
@magicDGS The point I was making was that using a different handler to get attributes like group name than was originally used to create the work unit isn't how Barclay itself gets them, so write the test this way ? There isn't anything that says that a handler needs to be stateless - the fact that this currently works with the default handler is completely accidental. It would be quite easy to make a perfectly valid change to the default handler that would cause this test to fail, but pass if the same handler was used.

The doc should probably be more explicit about this, but in the meantime it seems much more useful, and less brittle, to have a test that ensures that this method does the right thing when used as Barclay uses it.

Reply:
Done.